packages/genesys-spark-components/src/components/stable/gux-form-field/components/gux-form-field-radio-group/gux-form-field-radio-group.tsx

```diff
@@ -0,0 +1,139 @@
+import { Component, Element, h, JSX, State, Prop, Watch } from '@stencil/core';
+
+import { OnMutation } from '@utils/decorator/on-mutation';
+import { hasSlot } from '@utils/dom/has-slot';
+
+import {
+  GuxFormFieldError,
+  GuxFormFieldHelp,
+  GuxFormFieldLegendLabel,
+  GuxFormFieldFieldsetContainer
+} from '../../functional-components/functional-components';
+import { getSlotTextContent } from '@utils/dom/get-slot-text-content';
+
+import { trackComponent } from '@utils/tracking/usage';
+
+/**
+ * @slot group-label - Required slot for label tag
+ * @slot group-error - Optional slot for error message
+ * @slot group-help - Optional slot for help message
+ */
+@Component({
+  styleUrl: 'gux-form-field-radio-group.scss',
+  tag: 'gux-form-field-radio-group-beta',
+  shadow: true
+})
+export class GuxFormFieldRadioGroupBeta {
+  private disabledObserver: MutationObserver;
+  private label: HTMLLabelElement;
+
+  @Element()
+  private root: HTMLElement;
+
+  /**
+   *  Radio group has error text.
+   */
+  @State()
+  private hasGroupError: boolean = false;
+
+  /**
+   *  radio group has help text.
+   */
+  @State()
+  private hasGroupHelp: boolean = false;
+
+  /**
+   * Disables the radio buttons in the group.
+   */
+  @Prop()
+  disabled: boolean = false;
+
+  @Watch('hasGroupError')
+  watchGroupError(hasGroupError: boolean): void {
+    const radioSlots = this.root.querySelectorAll('gux-form-field-radio');
+    if (radioSlots) {
+      radioSlots.forEach(item => {
+        item.hasGroupError = hasGroupError;
```

Comment:
I was thinking that each individual radio slot prob doesn't need to know about the group so naming the prop `hasError` for its usage `item.hasError` would localize the disabled state to that individual item. Similar with help state but probably not for group since all the inputs seem tied together in that scenario. Thoughts?

Reply:
The idea is that individual radio buttons have their own error state, which is true or false depending on error slot being added to the individual radio button. Individual radio buttons need to know about the group error state because the error styling is added to each radio button if the group has an error slot. The group error and individual error state needs to be separate because of how on slot change works to update the state.